### ISSUE TITLE
Creatures can attack after move that ends in moat

### ISSUE THREAD

**Issue by HarkUhm** [vcmi/vcmi#2916]
**Describe the bug**
Creatures can attack a unit that stands on the other side of the moat just after they walked into the moat

**Expected behavior**
After stepping on a moat creatures shouldn't attack

**Version**
 - OS: [Windows 10]
 - Version: [VCMI 1.3.2]

**Comment by rilian-la-te** [comment]
How it behave in OH3? Can they attack, if they stop at moat? Especially flying creatures?

**Comment by dydzio0614** [comment]
BTW related: creatures should not be able to attack after moving to tile with quicksand, in H3 if quicksand is visible then highlighting unit that is covered from all direction in quicksand results in "circle" cursor rather than sword cursor

**Comment by rilian-la-te** [comment]
@dydzio0614  yes, look like "trap" property should not only stop creatures, but delay their attack too.

**Comment by HarkUhm** [comment]
OH:
![kjlk5](https://github.com/vcmi/vcmi/assets/141812373/7b885b9a-057a-4861-a6a4-c424c5d43f84)
VCMI:
![kjlk](https://github.com/vcmi/vcmi/assets/141812373/f112b8bf-2f82-4fa4-8ff5-d2ca7d136b99)
![kjlk5666](https://github.com/vcmi/vcmi/assets/141812373/6c4022b9-cbfc-4b6d-848c-10bf0353dafd)

**Comment by rilian-la-te** [comment]
@HarkUhm can you test it with normal SoD (without HD mod)? Anyway,  trap property need to be extended to cover this case (trap is used on quicksands and moats).

**Comment by HarkUhm** [comment]
Yes, without HDMod it's the same
